bootstraps/stage2/spack.in.yaml

```diff
@@ -0,0 +1,52 @@
+# This is a Spack Environment file.
+#
+# It describes a set of packages to be installed, along with
+# configuration settings.
+spack:
+  # add package specs to the `specs` list
+  specs:
+  - glibc %gcc@=13.2.0 arch=linux-spack-x86_64 ^linux-headers%gcc@12
```

Comment:
Can `%gcc@12` be dropped from `linux-headers` or is it installing stuff conditionally after inspecting the compiler?

Reply:
Linux-headers needs a fully-functional compiler, or at least a compiler with a libc, to build. It doesn’t seem to matter if it’s the host or target compiler though, so this is my solution to getting it to use the host compiler for just that one package. The version isn’t important, again it’s the only way I know to pin a specific compiler.

I’m now considering whether I could script it so that we generate a larger number of env files, and just generate an env whenever we want to change compilers to get rid of these version pins. Should work, but it will be more pieces.

Reply:
I switched this to using a requirement for a compiler version lower than the one we're compiling.  I don't like that solution, but nothing else I've tried has worked.